code/game/machinery/cloning.dm

```diff
@@ -434,49 +430,58 @@
 
 	if(!usr)
 		return
-	if(usr.stat != 0)
+	if(usr.incapacitated())
 		return
-	go_out(usr)
+	go_out()
 	add_fingerprint(usr)
-	return
 
-/obj/machinery/clonepod/proc/go_out(user)
+/obj/machinery/clonepod/proc/go_out()
+	if(locked)
+		return
+	countdown.stop()
+
 	if(mess) //Clean that mess and dump those gibs!
-		if(occupant)
-			return
 		mess = 0
 		gibs(loc)
 		playsound(loc, 'sound/effects/splat.ogg', 50, 1)
 		update_icon()
 		return
 
-	if(!(occupant))
-		to_chat(user, "<span class=\"warning\">The cloning pod is empty!</span>")
+	if(!occupant)
 		return
 
-	if(locked)
-		to_chat(user, "<span class=\"warning\">The cloning pod is locked!</span>")
-		return
+	if(grab_ghost_when == CLONER_MATURE_CLONE)
+		clonemind.transfer_to(occupant)
+		occupant.grab_ghost()
+		to_chat(occupant, "<span class='notice'><b>There is a bright flash!</b><br>\
+			<i>You feel like a new being.</i></span>")
+		occupant.flash_eyes()
 
-	if(occupant.client)
-		occupant.client.eye = occupant.client.mob
-		occupant.client.perspective = MOB_PERSPECTIVE
 	occupant.forceMove(get_turf(src))
 	eject_wait = 0 //If it's still set somehow.
 	domutcheck(occupant) //Waiting until they're out before possible notransform.
+	if(ishuman(occupant))
+		var/mob/living/carbon/human/H = occupant

```

Comment:
It's actually typed for `living`. I wonder if there were plans to be able to clone..well, anything.